### ISSUE TITLE
Unable to resize application from top edge when using `App.MainWindow.AppWindow.TitleBar.ExtendsContentIntoTitleBar = true`

### ISSUE THREAD

**Issue by harvinders** [microsoft/WindowsAppSDK#3694]
### Describe the bug

The issue emerges when we use `App.MainWindow.AppWindow.TitleBar.ExtendsContentIntoTitleBar = true; ` instead of `App.MainWindow.ExtendsContentIntoTitleBar = true; ` to extend the content into the title bar. 

However, using the latter option doesn't allow interaction elements in the titlebar, such as a button or a text box in the titlebar.

### Steps to reproduce the bug

1. Extend the content using `App.MainWindow.AppWindow.TitleBar.ExtendsContentIntoTitleBar = true; `
2. Try to resize the application using the top edge of the application window
3. The application is not resized, however, the application is moved. 

### Expected behavior

The application should be resized

### NuGet package version

Windows App SDK 1.3.2: 1.3.230602002

### Packaging type

Packaged (MSIX)

### Windows version

Windows 10 version 21H2 (19044, November 2021 Update)

### IDE

Visual Studio 2022

**Comment by DarranRowe** [comment]
I haven't fully messed around in all DPI values, but I have noticed that manually setting the drag rectangle with Y set at 5 or 6 pixels allows you to resize from the top edge.
Microsoft.UI.Windowing.AppWindowTitleBar.ExtendsContentIntoTitleBar acts like it doesn't take WS_THICKFRAME/WS_SIZEBOX into account on the top edge when setting the drag rectangle.

**Comment by zhuxb711** [comment]
Same issue here. @DarranRowe 's solution only works if use `Window.ExtendsContentIntoTitleBar` instead of `AppWindow.TitleBar.ExtendsContentIntoTitleBar`. However that would not behave as expected, the titlebar would lost the customization.

**Comment by DarranRowe** [comment]
@zhuxb711 
It is actually more interesting than that. It tends to be hard to see how weird the interaction between AppWindowTitleBar and Window is.
So let us look at a Windows API application.

https://github.com/microsoft/WindowsAppSDK/assets/52577874/ba74fa8e-723b-45c7-8fe8-a3d836d1acdb

This first obtains a Microsoft.UI.Windowing.AppWindow using the window Id, and then it gets a Microsoft.UI.Windowing.AppWindowTitleBar from the AppWindow using TitleBar. This should look familiar since it is the 1.2 or older method for WinUI applications which was made simpler with 1.3 by adding the AppWindow property. It then calls ExtendsContentIntoTitleBar. As you can see, the behaviour along the top edge is identical to what you would see when you are using WinUI 3.
If we then add the code:
```
std::array<winrt::Windows::Graphics::RectInt32, 1> drag_rect{};
auto left_inset = title_bar.LeftInset();
auto right_inset = title_bar.RightInset();
auto height = title_bar.Height();
auto window_size = m_appwindow.ClientSize();

drag_rect[0].X = left_inset;
drag_rect[0].Y = 6;
drag_rect[0].Width = window_size.Width - left_inset - right_inset;
drag_rect[0].Height = height;

title_bar.SetDragRectangles(drag_rect);
```

https://github.com/microsoft/WindowsAppSDK/assets/52577874/468f9a5c-425c-4db4-a7df-a0b0fc187754

The window resizes along the top edge with no issues. This application doesn't have an instance of Microsoft.UI.Xaml.Window, so it cannot be calling ExtendsContentIntoTitleBar on that, it is calling AppWindowTitleBar.ExtendsContentIntoTitleBar.

To show how weird this is in WinUI 3, first let's look at what happens if we just call ExtendsContentIntoTitleBar.
```
auto app_window = AppWindow();
auto title_bar = app_window.TitleBar();
title_bar.ExtendsContentIntoTitleBar(true);
```
I'm caching the AppWindow and AppWindowTitleBar instances since I will be making multiple calls on them. It is more efficient this way.

https://github.com/microsoft/WindowsAppSDK/assets/52577874/bb2e4af4-0bbd-416d-b502-eb8a3b738b2d

If you notice, this is behaving identically to how the Windows API application was. So, if we now try to fix the drag rectangle using the same code as above, this is where fun things happen.

https://github.com/microsoft/WindowsAppSDK/assets/52577874/b0d0f8b6-d16b-41fc-8091-e40435f575b5

This stops the top edge drag cursor from showing. While it not may be obvious, left click on the top few pixels doesn't drag the window, so the drag rectangle was set successfully. However, no resize happens. This is coming from the window that is created by WinUI 3.

So as a bit of a tl;dr.
The drag rectangle automatically set by AppWindowTitleBar.ExtendsContentIntoTitleBar is bad. It doesn't take resizable windows into account or it grabs input when it shouldn't. This does need to be fixed.
Changing the size of the drag rectangle does work. It allows input from the top edge of the window to get to the window below it.

If there are any problems with how this interacts with Microsoft.UI.Xaml.Window, then this would be a bug in WinUI 3 and would need to be addressed separately. It is technically outside the scope of this repository.

**Comment by zhuxb711** [comment]
@DarranRowe I totally agree with you. Top cursor do not showing and we could not resize it even we adjust the drag region.

**Comment by AndromedaMelody** [comment]
UWP in CoreWindow has this bug, too.
